src/dataset/resource.rs

```diff
@@ -23,15 +23,38 @@ pub enum Type {
     Pdf,
     Csv,
     JsonLd,
+    Xml,
+    Image,
+    Archive,
+    PlainText,
+}
+
+impl From<&'_ str> for Type {
+    fn from(val: &str) -> Self {
+        match val {
+            "text/plain" => Self::PlainText,
+            "application/pdf" => Self::Pdf,
+            "application/xml" => Self::Xml,
+            "image/png" => Self::Image,
+            "wms_xml" => Self::Image,
```

Comment:
Isn't this a Web Map Service? So I would not lump that together with images but either have a variant for mapping services or at least services.